packages/worker/src/tasks/syncPackageVulnerabilities.ts

```diff
@@ -0,0 +1,140 @@
+      skipUpdateIfNoValuesChanged: true,
+      conflictPaths: ['packageName', 'osvId'],
+    });
+    entityBuffer = [];
+  };
+
+  return async (osvData: OpenSourceVulnerability | null) => {
+    if (osvData === null) {
+      await flush();
+      return;
+    }
+
+    const affectedEntries = osvData.affected?.filter((it) => it.package.ecosystem === 'npm');
+    if (!affectedEntries?.length) {
+      return;
+    }
+
+    const affectedRangesByPackage: Record<string, string[]> = {};
+
+    for (const affectedEntry of affectedEntries) {
+      const affectedRanges = affectedEntry.ranges?.filter(
+        (it) => it.type === OSVAffectedRangeType.ECOSYSTEM
+      );
+      if (!affectedRanges) {
+        continue;
+      }
+
+      const affectedRangeParts = [];
+      for (const range of affectedRanges) {
+        const introducedVer = range.events.find((it) => !!it.introduced);
+        const fixedVer = range.events.find((it) => !!it.fixed);
+
+        if (introducedVer && fixedVer) {
+          affectedRangeParts.push(`>=${introducedVer.introduced} <${fixedVer.fixed}`);
+        }
+
+        if (introducedVer && !fixedVer) {
+          affectedRangeParts.push(`>=${introducedVer.introduced}`);
+        }
+      }
+
+      const affectedVersions = affectedEntry.versions;
+      if (affectedVersions) {
+        affectedRangeParts.push(...affectedVersions);
+      }
+
+      if (!affectedRangeParts.length) {
+        continue;
+      }
+
+      if (!affectedRangesByPackage[affectedEntry.package.name]) {
+        affectedRangesByPackage[affectedEntry.package.name] = affectedRangeParts;
+      } else {
+        affectedRangesByPackage[affectedEntry.package.name].push(...affectedRangeParts);
+      }
+    }
+
+    entityBuffer = entityBuffer.concat(
+      Object.entries(affectedRangesByPackage).map(([packageName, affectedRanges]) => ({
+        packageName,
+        packageVersionRange: affectedRanges.join(', '),
+        osvId: osvData.id,
+        osvData: osvData,
+      }))
+    );
+
+    if (entityBuffer.length >= batchSize) {
+      await flush();
+    }
+  };
+}
+
+export async function syncPackageVulnerabilities() {
+  const packageVulnerabilitySynchronizer = createPackageVulnerabilitySynchronizer();
+  const extractStream = TarStream.extract();
+
+  extractStream.on('entry', async (header, stream, next) => {
+    // Example path: advisory-database-main/advisories/github-reviewed/2022/05/GHSA-23wx-cgxq-vpwx/GHSA-23wx-cgxq-vpwx.json
+    const filePath = header.name;
+
+    const isReviewed = filePath.includes('advisories/github-reviewed');
+
+    if (!isReviewed || !filePath.endsWith('.json')) {
+      // Stream has to drained to continue
+      stream.on('end', () => next());
+      stream.resume();
+      return;
+    }
+
+    const fileBuffer = await drainStreamToBuffer(stream);
+    const fileContents = fileBuffer.toString();
+
+    const osvData = JSON.parse(fileContents) as OpenSourceVulnerability;
```

Comment:
Proposal: use some validation tool (e.g. https://github.com/gcanti/io-ts) on JSON responses from external resources across the project. It would give us some meaningful errors instead of 'access to property of undefined' in case of external API format changes. Not to be done in this PR, but in a separate task maybe.